numalogic/udfs/trainer/_base.py

```diff
@@ -345,10 +350,19 @@ def get_feature_arr(
             inf_counter: Number of inf values
         """
         nan_counter = 0
+        if max_value_map and len(max_value_map) != len(metrics):
+            raise ValueError("max_value_list and metrics list have different length")
         for col in metrics:
             if col not in raw_df.columns:
                 raw_df[col] = fill_value
                 nan_counter += len(raw_df)
+            if max_value_map:
+                raw_df.loc[raw_df[col] > max_value_map[col], col] = max_value_map[col]
```

Comment:
Why not use pd.clip?

Reply:
good suggestion! Never knew this existed!